Source/WebCore/page/ElementTargetingController.cpp

```diff
@@ -585,44 +650,143 @@ Vector<TargetedElementInfo> ElementTargetingController::findTargets(TargetedElem
     if (!view)
         return { };
 
-    RefPtr bodyElement = document->body();
-    if (!bodyElement)
+    static constexpr OptionSet hitTestOptions {
+        HitTestRequest::Type::ReadOnly,
+        HitTestRequest::Type::DisallowUserAgentShadowContent,
+        HitTestRequest::Type::IgnoreCSSPointerEventsProperty,
+        HitTestRequest::Type::CollectMultipleElements,
+        HitTestRequest::Type::IncludeAllElementsUnderPoint
+    };
+
+    HitTestResult result { LayoutPoint { view->rootViewToContents(pointInRootView) } };
+    document->hitTest(hitTestOptions, result);
+
+    return { copyToVector(result.listBasedTestResult()), result.innerNonSharedElement() };
+}
+
+static Element* searchForElementContainingText(ContainerNode& container, const String& searchText)
```

Comment:
Can this be `const ContainerNode& container`? Same question for some the helpers above, which take `Element` / `ContainerNode`.

Reply:
Hm... so `searchForElementContainingText` is non-const because the `makeRangeSelectingNodeContents` helper takes a non-const `Node`:

```
WEBCORE_EXPORT SimpleRange makeRangeSelectingNodeContents(Node&);
```

`collectDocumentElementsFromChildFrames`, `collectChildFrameIdentifiers`, and several other helper functions related to deriving targeted element selectors can both be const though — I'll make that change in this patch.